middle_end/compilenv.ml

```diff
@@ -141,6 +141,14 @@ let reset ?packname name =
       (Ident.create_persistent name)
       (current_unit_linkage_name ())
   in
+  Compilation_unit.set_current compilation_unit;
```

Comment:
I believe it is correct, but it might be useful to add a comment
explaining why we need the double registration. My understanding
is that the first one is necessary for parts shared with upstream
(e.g. in `Asmpackager`) while the second one is necessary for
flambda2-specific parts.

Reply:
Comment added